Added conversion support for kinect depth images.

carlosmccosta opened the pull request:
Added sensor_msgs::Image conversion to cv::Mat from rqt_image_view in order to be able to create videos from kinect color and depth images
(cv_bridge
currently doesn't support 16UC1 image encoding).

Related PR:
https://github.com/ros-perception/image_pipeline/pull/147

Code adapted from:
https://github.com/ros-visualization/rqt_common_plugins/blob/groovy-devel/rqt_image_view/src/rqt_image_view/image_view.cpp

carlosmccosta committed:
Added sensor_msgs::Image conversion to cv::Mat from rqt_image_view in
order to be able to create videos from kinect color and depth images
(cv_bridge
currently doesn't support 16UC1 image encoding).

Code adapted from:
https://github.com/ros-visualization/rqt_common_plugins/blob/groovy-devel/rqt_image_view/src/rqt_image_view/image_view.cpp

vrabaud commented:
I have not merged that as there are a few related issues. I'd rather have a cvtColorForDisplay function that would use the same rules from rqt_image_view and that would have a few parameters, of which range would be (but not necessarily for depth). I just need a bit of time but if you're willing to do it, thx !

carlosmccosta committed:
Improved efficiency by using toCvShare instead of toCvCopy.

carlosmccosta commented:
Thanks for the feedback :)
I refactored the code to use cv_bridge::toCvShare instead of cv_bridge::toCvCopy (for efficiency reasons) and changed the function to the requested name.
I also changed the docs and function parameters in order to not give the impression that the parameters values are for depth images only.

Have a nice day :)

vrabaud commented:
I was out for a month, sorry. I am not entirely satisfied with the result but I'll make another PR for discussion. Thx !
